demisto_sdk/commands/content_graph/parsers/pack.py

```diff
@@ -0,0 +1,174 @@
+            self.append(content_item)
+            return True
+        return False
+
+
+class PackContentItems:
+    """ A class that holds all pack's content items in lists by their types.
+    """
+    def __init__(self) -> None:
+        self.classifier = ContentItemsList(content_type=ContentType.CLASSIFIER)
+        self.correlation_rule = ContentItemsList(content_type=ContentType.CORRELATION_RULE)
+        self.dashboard = ContentItemsList(content_type=ContentType.DASHBOARD)
+        self.generic_definition = ContentItemsList(content_type=ContentType.GENERIC_DEFINITION)
+        self.generic_module = ContentItemsList(content_type=ContentType.GENERIC_MODULE)
+        self.generic_type = ContentItemsList(content_type=ContentType.GENERIC_TYPE)
+        self.incident_field = ContentItemsList(content_type=ContentType.INCIDENT_FIELD)
+        self.incident_type = ContentItemsList(content_type=ContentType.INCIDENT_TYPE)
+        self.indicator_field = ContentItemsList(content_type=ContentType.INDICATOR_FIELD)
+        self.indicator_type = ContentItemsList(content_type=ContentType.INDICATOR_TYPE)
+        self.integration = ContentItemsList(content_type=ContentType.INTEGRATION)
+        self.job = ContentItemsList(content_type=ContentType.JOB)
+        self.layout = ContentItemsList(content_type=ContentType.LAYOUT)
+        self.list = ContentItemsList(content_type=ContentType.LIST)
+        self.mapper = ContentItemsList(content_type=ContentType.MAPPER)
+        self.modeling_rule = ContentItemsList(content_type=ContentType.MODELING_RULE)
+        self.parsing_rule = ContentItemsList(content_type=ContentType.PARSING_RULE)
+        self.playbook = ContentItemsList(content_type=ContentType.PLAYBOOK)
+        self.report = ContentItemsList(content_type=ContentType.REPORT)
+        self.script = ContentItemsList(content_type=ContentType.SCRIPT)
+        self.test_playbook = ContentItemsList(content_type=ContentType.TEST_PLAYBOOK)
+        self.trigger = ContentItemsList(content_type=ContentType.TRIGGER)
+        self.widget = ContentItemsList(content_type=ContentType.WIDGET)
+        self.wizard = ContentItemsList(content_type=ContentType.WIZARD)
+        self.xsiam_dashboard = ContentItemsList(content_type=ContentType.XSIAM_DASHBOARD)
+        self.xsiam_report = ContentItemsList(content_type=ContentType.XSIAM_REPORT)
+
+    def iter_lists(self) -> Iterator[ContentItemsList]:
+        for attribute in vars(self).values():
+            yield attribute
+
+    def append(self, obj: ContentItemParser) -> None:
+        """ Appends a content item by iterating the content item lists
+        until finds the correct list and appends to it.
+
+        Args:
+            obj (ContentItemParser): The conten item to append.
+
+        Raises:
+            NotAContentItem: If did not find any matching content item list.
+        """
+        for content_item_list in self.iter_lists():
```

Comment:
Keep a dict of `ContentType -> ContentItemsList`, so this is o(1)

Reply:
O(|ContentType|) = O(1)
the idea is to avoid multiple static dicts of content types like we have now in constants (to ease the process of adding new items).

Reply:
No need to support a constant dict.
It can be built lazily at runtime, and kept, like a cache.